spectator/http_client.cc

```diff
@@ -233,6 +233,78 @@ HttpResponse HttpClient::do_post(const std::string& url,
   return HttpResponse{http_code, std::move(resp), std::move(resp_headers)};
 }
 
+HttpResponse HttpClient::Get(const std::string& url) const {
+  return do_get(url, 0);
+}
+
+HttpResponse HttpClient::do_get(const std::string& url,
+                                int attempt_number) const {
```

Comment:
This method has a lot of overlap with `do_post`, would it make sense to refactor so more of that common logic is reused?

Reply:
Yes, I was planning to do it soon after I release a version of the system agent that needs to use this GET functionality.